tools/twix/src/panels/image_color_select.rs

```diff
@@ -28,272 +31,393 @@ use crate::{
+
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Copy)]
 
 enum ImageKind {
     YCbCr422,
 }
-
-struct PixelColor {
-    red: f32,
-    green: f32,
-    blue: f32,
-}
-
-impl PixelColor {
-    pub const BLACK: Self = Self {
-        red: 0.0,
-        green: 0.0,
-        blue: 0.0,
-    };
-    pub const WHITE: Self = Self {
-        red: 1.0,
-        green: 1.0,
-        blue: 1.0,
-    };
+pub struct ImageColorSelectPanel {
+    nao: Arc<Nao>,
+    image_buffer: ImageBuffer,
+    field_color: ValueBuffer,
+    cycler_selector: VisionCyclerSelector,
+    brush_size: f32,
+    selection_mask: ColorImage,
+    x_axis: Axis,
+    y_axis: Axis,
+    filter_by_other_axes: bool,
 }
 
-struct Statistics {
-    max: PixelColor,
-    min: PixelColor,
-    average: PixelColor,
-    pixel_count: usize,
+struct ColorArray {
+    red: Vec<f32>,
+    green: Vec<f32>,
+    blue: Vec<f32>,
 }
 
-impl Default for Statistics {
+impl Default for ColorArray {
     fn default() -> Self {
         Self {
-            max: PixelColor::BLACK,
-            min: PixelColor::WHITE,
-            average: PixelColor::BLACK,
-            pixel_count: 0,
+            red: vec![0.0; 256],
+            green: vec![0.0; 256],
+            blue: vec![0.0; 256],
         }
     }
 }
 
-impl Statistics {
-    fn sample(mut self, pixel: Color32) -> Self {
-        let sum = pixel.r() as f32 + pixel.g() as f32 + pixel.b() as f32;
-        let mut pixel_color = PixelColor {
-            red: 0.0,
-            green: 0.0,
-            blue: 0.0,
-        };
-        if sum != 0.0 {
-            pixel_color.red = (pixel.r() as f32) / sum;
-            pixel_color.green = (pixel.g() as f32) / sum;
-            pixel_color.blue = (pixel.b() as f32) / sum;
+impl Add for ColorArray {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        fn element_wise_add(a: Vec<f32>, b: Vec<f32>) -> Vec<f32> {
+            a.iter().zip(b.iter()).map(|(a, b)| a + b).collect()
         }
 
-        self.max.red = self.max.red.max(pixel_color.red);
-        self.max.green = self.max.green.max(pixel_color.green);
-        self.max.blue = self.max.blue.max(pixel_color.blue);
-        self.min.red = self.min.red.min(pixel_color.red);
-        self.min.green = self.min.green.min(pixel_color.green);
-        self.min.blue = self.min.blue.min(pixel_color.blue);
-        self.average.red += pixel_color.red;
-        self.average.green += pixel_color.green;
-        self.average.blue += pixel_color.blue;
-        self.pixel_count += 1;
-        self
+        Self {
+            red: element_wise_add(self.red, rhs.red),
+            green: element_wise_add(self.green, rhs.green),
+            blue: element_wise_add(self.blue, rhs.blue),
+        }
     }
 }
 
-pub struct ImageColorSelectPanel {
-    nao: Arc<Nao>,
-    image_buffer: ImageBuffer,
-    cycler_selector: VisionCyclerSelector,
-    brush_size: f32,
+#[derive(Default)]
+struct Statistics {
```

Comment:
This type is unused and so is the `ColorArray` above, but clippy isn't smart enough to figure this out.

Reply:
done, also removed ImageKind